packages/react-combobox/src/components/ComboButton/useComboButtonStyles.ts

```diff
@@ -1,70 +1,201 @@
-    ...shorthands.borderRadius('4px'),
-    display: 'flex',
-    alignItems: 'center',
+    ...shorthands.border('1px', 'solid', 'transparent'),
+    ...shorthands.borderRadius(tokens.borderRadiusMedium),
+    boxSizing: 'border-box',
+    position: 'relative',
 
-    '&:focus-within': {
+    // windows high contrast mode focus indicator
+    ':focus-within': {
       outlineWidth: '2px',
       outlineStyle: 'solid',
       outlineColor: 'transparent',
-      borderBottomColor: '#0078d4',
-      borderBottomWidth: '2px',
     },
-  },
 
-  expandIcon: {
-    flexGrow: 0,
-    flexShrink: 0,
-    ...shorthands.padding('4px'),
+    // bottom focus border, shared with Input, Select, and SpinButton
+    '::after': {
+      boxSizing: 'border-box',
+      content: '""',
+      position: 'absolute',
+      left: '-1px',
+      bottom: '-1px',
+      right: '-1px',
+      height: `max(2px, ${tokens.borderRadiusMedium})`,
+      borderBottomLeftRadius: tokens.borderRadiusMedium,
+      borderBottomRightRadius: tokens.borderRadiusMedium,
+      ...shorthands.borderBottom('2px', 'solid', tokens.colorCompoundBrandStroke),
+      clipPath: 'inset(calc(100% - 2px) 0 0 0)',
+      transform: 'scaleX(0)',
+      transitionProperty: 'transform',
+      transitionDuration: motionDurations.ultraFast,
+      transitionDelay: motionCurves.accelerateMid,
+    },
+    ':focus-within:after': {
+      transform: 'scaleX(1)',
+      transitionProperty: 'transform',
+      transitionDuration: motionDurations.normal,
+      transitionDelay: motionCurves.decelerateMid,
+    },
+    ':focus-within:active:after': {
+      borderBottomColor: tokens.colorCompoundBrandStrokePressed,
+    },
   },
 
   content: {
-    flexGrow: 1,
-    backgroundColor: 'transparent',
-    ...shorthands.border('none'),
+    alignItems: 'center',
+    backgroundColor: tokens.colorTransparentBackground,
+    ...shorthands.border('0'),
+    boxSizing: 'border-box',
+    columnGap: horizontalSpacing.xxs,
+    display: 'flex',
+    justifyContent: 'space-between',
     textAlign: 'left',
-    ...shorthands.padding('4px', '8px'),
+    width: '100%',
 
     '&:focus': {
       outlineStyle: 'none',
     },
   },
+
+  // size variants
+  small: {
+    fontSize: tokens.fontSizeBase200,
+    lineHeight: tokens.lineHeightBase200,
+    ...shorthands.padding('3px', horizontalSpacing.sNudge),
+  },
+  medium: {
+    fontSize: tokens.fontSizeBase300,
+    lineHeight: tokens.lineHeightBase300,
+    ...shorthands.padding('5px', horizontalSpacing.mNudge),
+  },
+  large: {
+    columnGap: horizontalSpacing.sNudge,
+    fontSize: tokens.fontSizeBase400,
+    lineHeight: tokens.lineHeightBase400,
+    ...shorthands.padding('7px', horizontalSpacing.m),
+  },
+
+  // appearance variants
+  outline: {
+    backgroundColor: tokens.colorNeutralBackground1,
+    ...shorthands.border('1px', 'solid', tokens.colorNeutralStroke1),
+    borderBottomColor: tokens.colorNeutralStrokeAccessible,
+    color: 'red',
```

Comment:
Did you mean to leave color: 'red' in here or what that from debugging?

Reply:
definitely just a debugging artifact 🤣